adal/src/main/java/com/microsoft/aad/adal/TokenCacheAccessor.java

```diff
@@ -80,33 +86,60 @@ TokenCacheItem getATFromCache(final String resource, final String clientId, fina
         
         return accessTokenItem;
     }
-    
+
     /**
      * @return {@link TokenCacheItem} for regular token cache entry.  
      */
     TokenCacheItem getRegularRefreshTokenCacheItem(final String resource, final String clientId, final String user) {
+        final CacheEvent cacheEvent = startCacheTelemetryRequest(EventStrings.TOKEN_TYPE_RT);
+
         final String cacheKey = CacheKey.createCacheKeyForRTEntry(mAuthority, resource, clientId, user);
-        return mTokenCacheStore.getItem(cacheKey);
+        final TokenCacheItem item =  mTokenCacheStore.getItem(cacheKey);
+
+        if (item != null) {
+            cacheEvent.setTokenTypeRT(true);
+        }
+        Telemetry.getInstance().stopEvent(mTelemetryRequestId, cacheEvent, EventStrings.TOKEN_CACHE_LOOKUP);
+
+        return item;
     }
     
     /**
      * @return {@link TokenCacheItem} for MRRT token cache entry.  
      */
     TokenCacheItem getMRRTItem(final String clientId, final String user) {
+        final CacheEvent cacheEvent = startCacheTelemetryRequest(EventStrings.TOKEN_TYPE_MRRT);
         final String cacheKey = CacheKey.createCacheKeyForMRRT(mAuthority, clientId, user);
-        return mTokenCacheStore.getItem(cacheKey);
+        final TokenCacheItem item = mTokenCacheStore.getItem(cacheKey);
+
+        if (item != null) {
+            cacheEvent.setTokenTypeMRRT(true);
+            cacheEvent.setTokenTypeFRT(item.isFamilyToken());
+        }
+        Telemetry.getInstance().stopEvent(mTelemetryRequestId, cacheEvent, EventStrings.TOKEN_CACHE_LOOKUP);
```

Comment:
>Telemetry.getInstance().stopEvent(mTelemetryRequestId, cacheEvent, EventStrings.TOKEN_CACHE_LOOKUP); [](start = 8, length = 100)

If we do stop event here, can it give all the cache lookup sequence we have? including the fallback?